leshan-server-redis/src/main/java/org/eclipse/leshan/server/redis/RedisRegistrationStore.java

```diff
@@ -828,14 +828,22 @@ public void run() {
                         System.currentTimeMillis(), 0, cleanLimit);
 
                 for (byte[] endpoint : endpointsExpired) {
-                    Registration r = deserializeReg(j.get(toEndpointKey(endpoint)));
-                    if (!r.isAlive(gracePeriod)) {
-                        Deregistration dereg = removeRegistration(j, r.getId(), true);
-                        if (dereg != null)
-                            expirationListener.registrationExpired(dereg.getRegistration(), dereg.getObservations());
+                    byte[] regBytes = j.get(toEndpointKey(endpoint));
+                    if (regBytes == null) {
+                        LOG.warn("An expiration index points to an unknown registration: '{}'",
+                                new String(endpoint, UTF_8));
```

Comment:
I'm not sure but I guess this happens on some rare race condition where registration is removed at the same time that cleaner try to remove expiration ? 

If this is the reason, I guess this is a normal behavior so maybe a warn is to strong ?